app/bundles/LeadBundle/Services/AnonymizeContactCompanyData.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Mautic\LeadBundle\Services;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Mautic\EmailBundle\Model\EmailModel;
+use Mautic\FormBundle\Model\SubmissionModel;
+use Mautic\LeadBundle\Entity\Company;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Entity\LeadField;
+use Mautic\LeadBundle\Helper\AnonymizeHelper;
+use Mautic\LeadBundle\Model\FieldModel;
+use Psr\Log\LoggerInterface;
+
+class AnonymizeContactCompanyData
```

Comment:
```suggestion
final class AnonymizeContactCompanyData
```
I have a suggestion regarding renaming the service, `ContactCompanyDataAnonymizer`.